### PR TITLE
Updates to Kafka 3, Zipkin 3 and the latest build setup

### PR DESCRIPTION
**Pull request by codefromthecrypt**
Here's to more years of this project!

Java 17 smoothies thanks to @TeamModerne. We can use the [app](https://groups.google.com/g/zipkin-dev/c/F9J1U-3WMHI) for more after this.

```
$ mvn -U org.openrewrite.maven:rewrite-maven-plugin:run -Drewrite.recipeArtifactCoordinates=org.openrewrite.recipe:rewrite-migrate-java:RELEASE -Drewrite.activeRecipes=org.openrewrite.java.migrate.UpgradeToJava17
```

### CODE REVIEW

**Comment by codefromthecrypt** [comment]
@jeqo made an incredible build-bin/test so we know it works when the PR is green. I am so happy I was able to update this with no experience.

File: .github/workflows/create_release.yml [hunk 1]
```diff
@@ -11,21 +11,26 @@ on:
...
         with:
           # Prevent use of implicit GitHub Actions read-only token GITHUB_TOKEN. We don't deploy on
           # the tag MAJOR.MINOR.PATCH event, but we still need to deploy the maven-release-plugin master commit.
           token: ${{ secrets.GH_TOKEN }}
           fetch-depth: 1  # only need the HEAD commit as license check isn't run
+      - name: Setup java
+        uses: actions/setup-java@v4
+        with:
+          distribution: 'zulu'  # zulu as it supports a wide version range
+          java-version: '17'  # earliest LTS and last that can compile the 1.6 release profile.
```
Comment (anuraaga) [inline .github/workflows/create_release.yml:27]:
```suggestion
          java-version: '17'
```

My comment on the other PR was probably confusing, AFAIU, this comment isn't true on both counts for java 17?

**Review by anuraaga — APPROVED** [review]

File: .github/workflows/create_release.yml (hunk 1 above)
Comment (codefromthecrypt) [reply .github/workflows/create_release.yml]: sorry this was supposed to be `  # earliest LTS supported by Spring Boot 3` I'll fix the others!

**Comment by codefromthecrypt** [comment]
thanks for the look @anuraaga!

**Pull request merged**